compat/src/main/scala-2.11_2.12/scala/collection/compat/PackageShared.scala

```diff
@@ -45,8 +45,12 @@ private[compat] trait PackageShared {
   }
 
   implicit def genericCompanionToCBF[A, CC[X] <: GenTraversable[X]](
-      fact: GenericCompanion[CC]): CanBuildFrom[Any, A, CC[A]] =
-    simpleCBF(fact.newBuilder[A])
+      fact: GenericCompanion[CC]): CanBuildFrom[Any, A, CC[A]] = {
+    val builder = if (fact == Seq) new IdentityPreservingBuilder[A, Seq](Seq.newBuilder[A])
+                  else if (fact == LinearSeq) new IdentityPreservingBuilder[A, LinearSeq](LinearSeq.newBuilder[A])
```

Comment:
do we also need `immutable` variants? `GenSeq`?

would it then be more performant to use an identity hash map?

thoughts @julienrf?

Reply:
`GenSeq` is gone, right?

Reply:
good point

Reply:
I now do the variants `c.Seq`, `i.Seq`, `c.LinearSeq`, `i.LinearSeq`.